app/org/sagebionetworks/bridge/dynamodb/DynamoUpload2.java

```diff
@@ -0,0 +1,155 @@
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIgnore;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import org.joda.time.LocalDate;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.models.upload.Upload;
+import org.sagebionetworks.bridge.models.upload.UploadRequest;
+
+/**
+ * This DynamoDB table stores metadata for Bridge uploads. This class also defines global secondary indices, which can
+ * be used for more efficient queries.
+ */
+@DynamoDBTable(tableName = "Upload2")
+public class DynamoUpload2 implements DynamoTable, Upload {
+    private boolean complete;
+    private long contentLength;
+    private String contentMd5;
+    private String contentType;
+    private String filename;
+    private String healthCode;
+    private LocalDate uploadDate;
+    private String uploadId;
+    private Long version;
+
+    /** This empty constructor is needed by the DynamoDB mapper. */
+    public DynamoUpload2() {}
+
+    /** Construct a DDB Upload object from an upload request and healthcode. */
+    public DynamoUpload2(UploadRequest uploadRequest, String healthCode) {
+        complete = false;
+        contentLength = uploadRequest.getContentLength();
+        contentType = uploadRequest.getContentType();
+        contentMd5 = uploadRequest.getContentMd5();
+        filename = uploadRequest.getName();
+        this.healthCode = healthCode;
+        uploadId = BridgeUtils.generateGuid();
+    }
+
+    /** True if the upload is complete. False if not. */
+    @Override
+    public boolean isComplete() {
+        return complete;
+    }
+
+    /** @see #isComplete */
+    public void setComplete(boolean complete) {
+        this.complete = complete;
+    }
+
+    /** Upload content length in bytes. */
+    public long getContentLength() {
+        return contentLength;
+    }
+
+    /** @see #getContentLength */
+    public void setContentLength(long contentLength) {
+        this.contentLength = contentLength;
+    }
+
+    /** The base64-encoded, 128-bit MD5 digest of the object body. */
+    public String getContentMd5() {
+        return contentMd5;
+    }
+
+    /** @see #getContentMd5 */
+    public void setContentMd5(String contentMd5) {
+        this.contentMd5 = contentMd5;
+    }
+
+    /** MIME content type. */
+    public String getContentType() {
+        return contentType;
+    }
+
+    /** @see #getContentType */
+    public void setContentType(String contentType) {
+        this.contentType = contentType;
+    }
+
+    /** Name of the file to upload. */
+    @DynamoDBIndexHashKey(attributeName = "filename", globalSecondaryIndexName = "filename-index")
```

Comment:
Maintaining an index can be expensive.  If not desperately needed, I would suggest not add too many indices on tables directly supporting transactions.  On the other hand, once the data is exported, tables there are to support queries.  Indices should be set up there, downstream of the data pipeline.